Reference_Manual.md

```diff
@@ -3287,6 +3294,245 @@ This typed_data allows you to retrieve the data type of the underlying binary to
+
+``` C++
+using namespace hazelcast::client::sql;
+
+auto hz = hazelcast::new_client().get();
+
+// populate the map with some data
+auto map = hz.get_map("integers").get();
+for (int i = 0; i < 1000; ++i) {
+    map->put(i, i).get();
+}
+
+auto sql = hz.get_sql();
+// Create mapping for the integers. This needs to be done only once per map.
+auto result = sql.execute(R"(
+                CREATE OR REPLACE MAPPING integers
+                    TYPE IMap
+                    OPTIONS (
+                        'keyFormat' = 'int',
+                        'valueFormat' = 'int'
+                        )
+                    )").get();
+```
+### 7.11.3 Read SELECT Results
+`SELECT` results are fetched page by page. A page can accomodate zero or more rows.
+
+`sql_service::execute` executes the query and returns a `boost::future<std::shared_ptr<sql_result>>`. Via `sql_result::iterator()` method, `sql_result::page_iterator` is acquired and pages can be iterated.
+``` C++
+using namespace hazelcast::client::sql;
+
+auto hz = hazelcast::new_client().get();
+
+auto result = hz.get_sql().execute("SELECT * FROM integers").get();
+
+result->row_metadata(); // Tells about metadata
+
+for (auto itr = result->iterator(); itr.has_next();)
+{
+    auto page = itr.next().get();
+
+    for(const sql_page::sql_row& row : page->rows())
+    {
+        std::cout << row.get_object<int>(0)
+                  << ", "
+                  << row.get_object<int>(1)
+                  << '\n';
+    }
+}
+```
+
+**Note-1**: First call to `sql_result::page_iterator::next()` function will return `ready` future because first page is already loaded.
+``` C++
+auto itr = result->iterator();
+itr.next().get(); // This will not block even `get()` 
+	          // is called because it will return
+	          // a future which is in ready state.
+```
+**Note-2**: After first page every `sql_result::page_iterator::next()` call fetches a new page and it must not be called consecutively.
+``` C++
+auto itr = result->iterator();
+itr.next().get(); // It will return first page
+auto page_2 = itr.next(); // Made request to second page
+// page_2.get(); // I should have waited but didn't
+auto page_3 = itr.next(); // This probably throws an `hazelcast::illegal_access` exception
+                          // because previous page might not be fetched yet. To prevent this
+	                  // `page_2.get()` should have called to ensure that page_2 is fetched.
+```
+**Note-3**: If fetched page is marked as `last` there should not be any further fetch requests.
+``` C++
+auto itr = result->iterator();
+
+auto page_1 = itr.next().get();
+
+if (page_1->last())
+    itr.next(); // This will throw an `hazelcast::no_such_element` exception.
+```
+**Note-4**: Zero rows do not mean that it is the last page. `SELECT * FROM TABLE(generate_stream(1))` query mostly returns first page as empty because it is a stream.
```

Comment:
```suggestion
**Note-4**: Zero rows do not mean that it is the last page. `SELECT * FROM TABLE(generate_stream(1))` query mostly returns first page as empty because it is a stream. Stream is an endless flow of data from server to the client.
```